src/ux/table.ts

```diff
@@ -0,0 +1,321 @@
+    } else if (!this.options.extended) {
+      // show extended columns/properties
+      this.columns = this.columns.filter((c) => !c.extended);
+    }
+
+    this.data = rows;
+  }
+
+  public display(): void {
+    const { data, options } = this;
+    // column truncation
+    //
+    // find max width for each column
+    const columns = this.columns.map((c) => {
+      const maxWidth = Math.max(sw('.'.padEnd(c.minWidth - 1)), sw(c.header), getWidestColumnWith(data, c.key)) + 1;
+      return {
+        ...c,
+        maxWidth,
+        width: maxWidth,
+      };
+    });
+
+    // terminal width
+    const maxWidth = stdtermwidth - 2;
+    // truncation logic
+    const shouldShorten = (): void => {
+      // don't shorten if full mode
+      if (options['no-truncate'] ?? (!process.stdout.isTTY && !process.env.CLI_UX_SKIP_TTY_CHECK)) return;
+
+      // don't shorten if there is enough screen width
+      const dataMaxWidth = sumBy(columns, (c) => c.width);
+      const overWidth = dataMaxWidth - maxWidth;
+      if (overWidth <= 0) return;
+
+      // not enough room, short all columns to minWidth
+      for (const col of columns) {
+        col.width = col.minWidth;
+      }
+
+      // if sum(minWidth's) is greater than term width
+      // nothing can be done so
+      // display all as minWidth
+      const dataMinWidth = sumBy(columns, (c) => c.minWidth);
+      if (dataMinWidth >= maxWidth) return;
+
+      // some wiggle room left, add it back to "needy" columns
+      let wiggleRoom = maxWidth - dataMinWidth;
+      const needyCols = columns
+        .map((c) => ({ key: c.key, needs: c.maxWidth - c.width }))
+        .sort((a, b) => a.needs - b.needs);
+      for (const { key, needs } of needyCols) {
+        if (!needs) continue;
+        const col = columns.find((c) => key === c.key);
+        if (!col) continue;
+        if (wiggleRoom > needs) {
+          col.width = col.width + needs;
+          wiggleRoom -= needs;
+        } else if (wiggleRoom) {
+          col.width = col.width + wiggleRoom;
+          wiggleRoom = 0;
+        }
+      }
+    };
+
+    shouldShorten();
+
+    // print table title
+    if (options.title) {
+      options.printLine(options.title);
+      // print title divider
+      options.printLine(
+        ''.padEnd(
+          columns.reduce((sum, col) => sum + col.width, 1),
+          '='
+        )
+      );
+
+      options.rowStart = '| ';
+    }
+
+    // print headers
+    if (!options['no-header']) {
+      let headers = options.rowStart;
+      for (const col of columns) {
+        const header = col.header;
+        headers += header.padEnd(col.width);
+      }
+
+      if (headers) options.printLine(ansis.bold(headers));
+
+      // print header dividers
+      let dividers = options.rowStart;
+      for (const col of columns) {
+        const divider = ''.padEnd(col.width - 1, '─') + ' ';
+        dividers += divider.padEnd(col.width);
```

Comment:
if dividers really can be `undefined` like the type says, then `undefined += divider.padEnd(col.width)` would make a string starting with undefined.

Reply:
Same question here.